ee/clickhouse/queries/funnels/funnel_correlation.py

```diff
@@ -160,7 +160,7 @@ def get_partial_event_contingency_tables(self) -> List[EventContingencyTable]:
             FROM events AS event
             
             JOIN person_distinct_id AS pdi
-                ON pdi.distinct_id = events.distinct_id
+                ON pdi.distinct_id = events.distinct_id AND pdi.team_id = event.team_id
```

Comment:
Err, actually, good to resolve this here: I realised we're not following the right pattern for joining `person_distinct_id`s here:

```
JOIN (
                                        SELECT distinct_id,
                                               argMax(person_id, _timestamp) as person_id
                                          FROM (
                                                SELECT distinct_id,
                                                       person_id,
                                                       max(_timestamp) as _timestamp
                                                  FROM person_distinct_id
                                                 WHERE team_id = 2
                                                 GROUP BY person_id,
                                                          distinct_id,
                                                          team_id
                                                HAVING max(is_deleted) = 0
                                               )
                                         GROUP BY distinct_id
                                       ) as person_distinct_id
                                    ON person_distinct_id.distinct_id = events.distinct_id
```